### ISSUE TITLE
Not able to create container on BananaPi

### ISSUE THREAD

**Issue by kovu94** [michelde/openv-vcontrold-docker#3]
I recently got the ir-reader for my Viessmann heating system and tried to get the data to Home Assistant.
MQTT is the way I need to go as I have a docker instance of Home Assistant running where no Add-Ons are working.

The BananaPi is in the heating room, a x86-Server running other containers in another room.

I installed docker and portainer on the BananaPi successfully but when I try to deploy the stack for vcontrold I get the following error:

**_docker: no matching manifest for linux/arm/v7 in the manifest list entries._**

I'm using Arnbian 23.02.2 Jammy with Linux 5.15.93-sunxi Kernel.

Is it possible to run an armV7 container of vcontrold? I successfully compiled an started the original version but this version don't have MQTT as far as I see.

**Comment by michelde** [comment]
I think if you build the container on your BananaPi it should be compiled with your current architecture and you should be able to start the image. But you can also try to add the tag arm32v7 to the base image of debian-slim `FROM arm32v7/debian:stable-slim`

**Comment by kovu94** [comment]
Hey, thanks for the reply.
If I try to make a Stack with Portainer the following error occurs during deployment:

> failed to deploy a stack: vcontrold Pulling vcontrold Error Error response from daemon: pull access denied for michelmu/vcontrold-openv, repository does not exist or may require 'docker login': denied: requested access to the resource is denied 

Then I downloaded the code and replaced the string in it to FROM arm32v7/debian:stable-slim.
During the deployment of the image a lot of apt-get and installations are working in the background but it stops with an error:

> Step 6/9 : ADD config /docker/vcontrold/config/
ADD failed: file not found in build context or excluded by .dockerignore: stat config: file does not exist

I think this is related to the config file. I tried default and a link to my docker-data directory.

**Comment by michelde** [comment]
The build fails because you don't have the config folder. I would suggest to clone my repo and then do the docker build process on the command line.

**Comment by kovu94** [comment]
Looks good for now - I can't test at the moment because the reader isn't connected and I'm not at home. But build was okay:

> [+] Building 415.9s (12/12) FINISHED
 => [internal] load build definition from Dockerfile                                                                          0.2s
 => => transferring dockerfile: 538B                                                                                          0.1s
 => [internal] load .dockerignore                                                                                             0.4s
 => => transferring context: 2B                                                                                               0.0s
 => [internal] load metadata for docker.io/arm32v7/debian:stable-slim                                                         0.0s
 => [1/7] FROM docker.io/arm32v7/debian:stable-slim                                                                           0.1s
 => [internal] load build context                                                                                             0.3s
 => => transferring context: 214B                                                                                             0.0s
 => CACHED [2/7] RUN apt-get update &&     apt-get upgrade -y                                                                 0.0s
 => [3/7] RUN apt-get install -y build-essential vim subversion automake autoconf libxml2-dev mosquitto-clients git cmake   282.5s
 => [4/7] RUN mkdir openv && cd openv && git clone https://github.com/openv/vcontrold.git vcontrold-code                      5.2s
 => [5/7] RUN cd /openv && cmake ./vcontrold-code -DVSIM=ON -DMANPAGES=OFF &&     make &&     make install                   31.2s
 => [6/7] ADD config /etc/vcontrold/                                                                                          1.0s
 => [7/7] ADD startup.sh /                                                                                                    0.6s
 => exporting to image                                                                                                       94.1s
 => => exporting layers                                                                                                      94.0s
 => => writing image sha256:2dd734eb42efbc5b3bb029d374174172c51cccb58ab70b5f88f46a3dd8df1641                                  0.0s

Building stack now also worked (after excluding USB-device) and container is up and running.

I got some messages in logs but maybe they are related to missing reader? What about the last line about mqtt?

> Device /dev/vitocal
chmod: cannot access '/dev/vitocal': No such file or directory
vcontrold gestartet (PID 10)
MQTT: aktiv (var = true)
Aktualisierungsintervall: 30 sec
Lese Parameter: getTempWWObenIst,getTempWWsoll,getNeigungHK1,getTempVL,getTempRL,getPumpeStatusZirku,getBetriebArtHK1,getTempVListHK1,getTempRListHK1,getStatusVerdichter,getJAZ,getJAZHeiz,getJAZWW,getTempA,getPumpeStatusHK1
/startup.sh: 24: /config/mqtt_sub.sh: not found

One thing - I tried to manually import your image from dockerhub but this also doesn't work.
![Screenshot 2023-03-30 101103](https://user-images.githubusercontent.com/5141666/228781978-02a4678f-4fb5-4222-89fd-2ca8a3935912.png)
Maybe you can add the tag "arm" to the project then it could work without manually steps? Because the ludoviclehmann/vcontrold
image is downloading fine.

**Comment by kovu94** [comment]
Some more errors appearing:

(((SRV ERR: command unknown
/startup.sh: 27: /config/mqtt_publish.sh: not found))) --> solved. manually copied both files from cloned project.

**Comment by kovu94** [comment]
It's working now - I can connect to the controller using commands and also see MQTT messages in MQTT-Explorer.
Thank you very much for helping so far.

Your commands in expamle are not working so I connected to the controller with vclient -h 127.0.0.1 -p 3002 -c "command" and read a list of possible commands.
Some commands are working like getLZSNH or getStatusStoerung but a lot of temperature values are at -0,1°C
The controller is a Viessmann Vitotronic 200 Typ KW2 and I selected this one in the vcontrol.xml.

Do I have to replace the xml files?
